fs/sync.go

```diff
@@ -302,20 +302,26 @@ func (s *syncCopyMove) pairChecker(in ObjectPairChan, out ObjectPairChan, wg *sy
 			// Check to see if can store this
 			if src.Storable() {
 				if NeedTransfer(pair.dst, pair.src) {
-					// If destination already exists, then we must move it into --backup-dir if required
-					if pair.dst != nil && s.backupDir != nil {
-						remoteWithSuffix := pair.dst.Remote() + s.suffix
-						overwritten, _ := s.backupDir.NewObject(remoteWithSuffix)
-						err := Move(s.backupDir, overwritten, remoteWithSuffix, pair.dst)
-						if err != nil {
-							s.processError(err)
+					// If files are treated as immutable, fail if destination exists and does not match
+					if Config.Immutable && pair.dst != nil {
+						Errorf(pair.dst, "Immutable file modified")
```

Comment:
I think this log message needs a bit more explanation - how about this?

    Source and destination exist but don't match: immutable file modified